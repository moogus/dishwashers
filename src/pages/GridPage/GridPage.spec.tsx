import React from "react";
import { render } from "@testing-library/react";
import { mocked } from "ts-jest/utils";

import useLoadGridItems from "../../hooks/grid-items/load-grid-items";
import useGetGridItems from "../../hooks/grid-items/get-grid-items";

import { GridPage } from "./index";

jest.mock("../../hooks/grid-items/load-grid-items");
jest.mock("../../hooks/grid-items/get-grid-items");
jest.mock("@reach/router");

describe("<GridItem />", () => {
  const renderContainer = () =>
    render(<GridPage path="foo-path (needed for reach router)" />);

  describe("when product-grid api is loading", () => {
    beforeEach(() => {
      mocked(useLoadGridItems).mockClear();
      mocked(useGetGridItems).mockClear();
      mocked(useLoadGridItems).mockReturnValue({
        loading: true,
        error: false,
      });
      mocked(useGetGridItems).mockReturnValue([]);
    });

    test("a loading message is displayed", () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });

  describe("when product-grid api has loaded", () => {
    beforeEach(() => {
      mocked(useLoadGridItems).mockClear();
      mocked(useGetGridItems).mockClear();
      mocked(useLoadGridItems).mockReturnValue({
        loading: false,
        error: false,
      });
      mocked(useGetGridItems).mockReturnValue([
        {
          productId: "1234",
          price: "Â£1000",
          title: "What the dickens",
          image: "foo",
        },
      ]);
    });

    test("product grid page is rendered as expected", () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });
});
