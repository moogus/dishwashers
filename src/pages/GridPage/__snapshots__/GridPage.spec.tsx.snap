// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItem /> when product-grid api has loaded product grid page is rendered as expected 1`] = `
<div>
  <div
    class="sc-kEjbxe jllcuK"
  >
    <header>
      <h1
        class="sc-pFZIQ imiWkr pt-5 pb-5"
      >
        Dishwashers (114)
      </h1>
    </header>
    <main>
      <section>
        <div
          class="sc-iqHYGH hIPneH row"
        >
          <div
            class="sc-bdfBwQ gQwMDF col-md-3"
            data-testid="clickable-grid-item"
          >
            <div
              class="sc-gsTCUz wFoFg"
            >
              <div
                class="sc-dlfnbm gYlIvC row"
              >
                <div
                  class="sc-hKgILt bEKIEI col"
                >
                  <img
                    alt="What the dickens"
                    class="sc-eCssSg knjoeH"
                    src="foo"
                  />
                </div>
              </div>
              <div
                class="sc-jSgupP gylmqB"
              >
                <p
                  class="sc-gKsewC ebGEqF"
                >
                  What the dickens
                </p>
              </div>
              <div
                class="sc-iBPRYJ kspsru"
              >
                <p
                  class="sc-fubCfw gkHLwA"
                >
                  Â£1000
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</div>
`;

exports[`<GridItem /> when product-grid api is loading a loading message is displayed 1`] = `
<div>
  <div
    class="sc-jrAGrp cCayON"
  >
    Loading Dishwashers
  </div>
</div>
`;
